Apps/phhackerone/hackerone_connector.py

```diff
@@ -515,27 +533,58 @@ def _get_updated_reports(self, param, action_result):
             action_result.add_exception_details(err)
             return action_result.set_status(phantom.APP_ERROR, 'Failed to get reports')
 
+    def _extract_urls(self, source):
+        artifacts = []
+        url_pattern = re.compile(r'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+#]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+')
+        domain_pattern = re.compile(r'^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n?]+)')
+        invalid_pattern = re.compile(r'[^a-zA-Z0-9\.\-]')
+        for value in re.finditer(url_pattern, source):
+            url = value.group(0)
+            domain = None
+            for match in re.finditer(domain_pattern, url):
+                domain = match.group(1)
+                break
+            if domain and len(re.findall(invalid_pattern, domain)) == 0:
+                artifact = {}
+                artifact['label'] = 'extracted url'
+                artifact['name'] = 'extracted url'
+                artifact['source_data_identifier'] = 'Hacker1 url - {0}'.format(url[:60])
+                artifact['severity'] = 'medium'
+                cef = {}
+                cef['requestURL'] = url
+                cef['domain'] = domain
+                artifact['cef'] = cef
+                artifacts.append(artifact)
+        return artifacts
+
     def _test(self, action_result, param):
-        self.__print('_test()')
+        self.debug_print('_test()')
         try:
             config = self.get_config()
             url_params = {'filter[program][]': self._handle_unicode_for_input_str(config['program_name']), 'page[size]': 1}
             reports, _ = self._get_rest_data('https://api.hackerone.com/v1/reports', url_params)
             if reports:
-                self.__print('Successfully connected to HackerOne')
-                return action_result.set_status(phantom.APP_SUCCESS, 'Successfully connected to HackerOne')
+                return action_result.set_status(phantom.APP_SUCCESS, 'Test connectivity passed')
             else:
-                self.__print('Failed to connect to HackerOne')
-                return action_result.set_status(phantom.APP_ERROR, 'Failed to connect to HackerOne')
+                return action_result.set_status(phantom.APP_ERROR, 'Test connectivity failed')
         except Exception as e:
             err = self._get_error_message_from_exception(e)
-            self.__print('Failed to connect to HackerOne')
             action_result.add_exception_details(err)
-            return action_result.set_status(phantom.APP_ERROR, 'Failed to connect to HackerOne')
+            return action_result.set_status(phantom.APP_ERROR, 'Test connectivity failed')
 
     def _on_poll(self, param):
         self.__print('_on_poll()')
-        login_url = BaseConnector._get_phantom_base_url()
+        self.debug_print("There might be timezone variance. Please check for the timezone variance.")
+        current_time_marker = datetime.datetime.now()
```

Comment:
@dallanwagz, Please make sure that if you want to use the current UTC time, then use datetime.datetime.utcnow() method. Because the .now() method will return the current local time and not the UTC time.